components/crud-web-apps/common/backend/kubeflow/kubeflow/crud_backend/csrf.py

```diff
@@ -0,0 +1,108 @@
+"""
+Cross Site Request Forgery Blueprint.
+
+This module provides a Flask blueprint that implements protection against
+request forgeries from other sites. Currently, it is only meant to be used with
+an AJAX frontend, not with server-side rendered forms.
+
+The module implements the following protecting measures against CSRF:
+- Double Submit Cookie.
+- Custom HTTP Headers.
+- SameSite cookie attribute.
+
+To elaborate, the `Double Submit Cookie` procedure looks like the following:
+1. Browser requests `index.html`, which contains the compiled Javascript.
+2. Backend sets the `CSRF_COOKIE` by calling `set_cookie`. If the cookie
+   already exists, `set_cookie` overrides it with a new one. The cookie
+   contains a random value.
+3. Frontend (`index.html`) is loaded and starts making requests to the backend.
+   For every request, the frontend reads the `CSRF_COOKIE` value and adds a
+   `CSRF_HEADER` with the same value.
+4. Backend checks that the value of `CSRF_COOKIE` matches the value of
+   `CSRF_HEADER`. All endpoints are checked, except the index endpoint and
+   endpoints with safe methods (GET, HEAD, OPTIONS, TRACE).
+
+Custom Headers (`CSRF_HEADER`) provide an extra layer of protection, as
+cross-origin requests cannot include custom headers (assuming CORS is not
+misconfigured) because of the Same-Origin policy.
+
+The SameSite cookie attribute provides another layer of protection, but may
+impede usability so it is configurable. This attribute controls whether a
```

Comment:
The SameSite attribute is currently not configurable. Should we change the docstring to reflect that? I see that you've added a note that we may need to make it configurable in the future.

Reply:
Lets make it configurable then, it should be a small change.

I'll make it look for a `CSRF_SAMESITE` env var with a default value to `Strict`. If the value provided by the user is not `Strict`, `Lax` or `None` then again it will default to `Strict`
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#SameSite

Reply:
force pushed. The PR should only contain one commit responsible for the CSRF functionality.
I also made the `SameSite` attribute of the cookie configurable via the `CSRF_SAMESITE` variable.

I also think a good next step would be to document the ENV Vars that each web app uses in the respective READMEs. This will make clear to the users how they can configure each app.

Reply:
That's great! One question: maybe it's better to start the env var section in the README now and document this specific settings with this PR, which introduces it. You can add a note that the section is a work in-progress so that readers don't get confused. What do you think?

Reply:
ACK! I modified the README of the common code.
I also created #5483 as an umbrella issue for the web apps